src/site/en/blog/2020-07-09-no-to-routers/index.md

```diff
@@ -0,0 +1,246 @@
+
+And the last one – on the web frontend we sometimes have **"virtual routes"**. This means, we have different screen states for the same path. Because, well, you filled the first two steps of that wizard form – so we need to show you the third one and not allow you to go to the fourth one.
+
+# Why none of the popular routers solve the problem?
+
+For some unknown reason all of the popular routing solutions for web frontend focus on the tip of the iceberg, while making some significant mistakes in core architecture design.
+
+### Mistake #1: defining routes in view layer
+
+<figure>
+    <img
+        src="off-label.jpg"
+        alt='"Off-label" by Anna Zarubey'
+        srcset="off-label@2x.jpg"
+        width="1200"
+        height="675"
+    />
+    <figcaption>"Off-label" by <a href="https://t.me/anna_zarubey">Anna Zarubey</a></figcaption>
+</figure>
+
+As you already know, routing process is heavily dependent on business logic. The only two cases when routing and view should collide are mapping resolved state to page and rendering links.
+
+So there is no actual reason use your view logic for routes definition. And when you do something without reason, you make it difficult to understand and maintain.
+
+It still works quite good on small apps though, because they don't have any complex or asynchronous business logic, and the only thing they need is a list of route-page pairs.
+
+### Mistake #2: routing as simple mapping from paths to pages
+
+<figure>
+    <img
+        src="obvious.jpg"
+        alt='"Obvious" by Anna Zarubey'
+        srcset="obvious@2x.jpg 2x"
+        width="1200"
+        height="675"
+    />
+    <figcaption>"Obvious" by <a href="https://t.me/anna_zarubey">Anna Zarubey</a></figcaption>
+</figure>
+
+Router docs tell you can define routing in declarative style. Meaning the whole routing problem is just a key-value dictionary.
+
+No, it's not. It could be, but only in simple hello-world-ish cases, which get complex as soon as your app becomes 1 month old.
+
+In general it is a fully-fledged process with dependency loading, data processing and decision making. It is also full of side-effects: from external dependencies and data loading to browser history management.
+
+### Mistake #3: immediate transitions
+
+Let's assume, you are on a simple website with no javascript at all. When you click the link, are you immediately transitioned to your destination? No, even in this simple case you have to wait until the next page is loaded.
+
+Waiting for transitions is in the DNA of the web from day one. Users got used to waiting after clicking the link, they **expect the next page to be loaded.** This means, it's ok to wait, because it is going to show me my entire friend list, and that's a lot of data, and I'm on 2G internet in the middle of nowhere, so I totally understand.
```

Comment:
Получилась тавтология с `users got used` :–( Может так?

```suggestion
Waiting for transitions is in the DNA of the web from day one. We got used to waiting after clicking the link, and we **expect the next page to be loaded.** This means it's ok to wait because I requested my entire friend list, and that's a lot of data, and I'm on 2G internet in the middle of nowhere, so I totally understand.
```